src/profiler/profiler.h

```diff
@@ -1149,8 +1158,20 @@ struct ProfileOperator : public ProfileEvent {
     : ProfileEvent(name)
       , as_task_(name, &domain_)
       , name_(name)
-      , attributes_(attributes) {
-    SetCategories(domain_.name());
+      , attributes_(attributes)
+      , profiling_operator(true)
+      , profiling_task(true) {
+    if (strcmp(name, "CustomOperator") == 0) {
+      profiling_operator = false;
+      profiling_task = false;
+    } else if (strcmp(name, "Custom") == 0) {
+      profiling_task = false;
+    } else if (std::string(name).find("::") != std::string::npos) {
+      as_task_.setDomain(&custom_op_domain);
+      SetCategories(custom_op_domain.name());
```

Comment:
ideally you won't  need this else if logic. when engine sets the display name the correct display name is automatically. For example if my custom ops MyCustomOp(CustomOp) and MyCustomOp2(CustomOp):
then profiling result should return MyCustomOp::elemwise_add
MyCustomOp2::elemwise_add

Reply:
Yes the names are correct. But we still need to set the domain here to "Custom Operator" if the operator turns out to be a "sub-operator", that is, an operator called by a custom operator

Reply:
Given there are so many confusion about the name, it's better to write a function (with comment) to explain the logic instead of `strstr(name, "::")`

Reply:
Now there is a dedicated function